### PR TITLE
Port tests for triage/summarize.py

### PR DESCRIPTION
**Pull request by michaelkolber**
This goes with #18273. It currently relies on previous commits from #18273, but will be rebased onto master once that PR is merged.

/cc bentheelder spiffxp

### CODE REVIEW

**Comment by michaelkolber** [comment]
Actual additions for this PR start from `Add tests for cluster.go, output.go and text.go` (currently 110a10f)

File: triage/summarize/cluster.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Contains functions that cluster failures.
```
Comment (BenTheElder) [inline triage/summarize/cluster.go:18]: usually the package comment is just on one file, often a separate `doc.go`

File: triage/summarize/text_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		name     string
+		argument []string
+		want     []int
+	}{
+		{"Exact match", []string{"an exact match", "an exact match"}, []int{14}},
+		{"Replaced word", []string{"some example string", "some other string"}, []int{5, 7, 7}},
+		{"Deletion", []string{"a problem with a common set", "a common set"}, []int{2, 7, 1, 4, 13}},
+	}
+
+	for _, tc := range testCases {
```
Comment (BenTheElder) [inline triage/summarize/text_test.go:95]: if these are expensive you may want t.Parallel() usage

**Comment by BenTheElder** [comment]
quick scan: tests looking pretty good. minor nits and CI failures. did not review the previous commit (Except one nit, whoops)

File: triage/summarize/cluster.go (hunk 1 above)
Comment (michaelkolber) [reply triage/summarize/cluster.go:18]: This isn't a package comment, it's just describing what the functions in that file do. I intentionally left a blank line between it and the package declaration.

**Comment by michaelkolber** [comment]
Rebasing onto current tip of #18273

**Comment by BenTheElder** [comment]
needs rebase

**Comment by michaelkolber** [comment]
/retest
Looks like a connection issue

**Comment by spiffxp** [comment]
ref: https://github.com/kubernetes/test-infra/issues/18726

**Comment by michaelkolber** [comment]
Rebased onto master

**Comment by michaelkolber** [comment]
Rebased onto master to get #18772 in history

**Comment by michaelkolber** [comment]
Rebase onto master so that history includes #18772
/retest

**Comment by BenTheElder** [comment]
note that as long as there aren't merge conflicts prow is already going to merge in this PR prior to testing so rebasing is really only necessary to manually resolve conflicts

**Comment by michaelkolber** [comment]
Oh, I thought it doesn't auto-merge if tests fail, and since tests were failing I added the necessary code

**Comment by BenTheElder** [comment]
> Oh, I thought it doesn't auto-merge if tests fail, and since tests were failing I added the necessary code

it won't, but if you merge fixes into the repo then just running the tests again is sufficient, you don't actually have to rebase (which will drop your lgtm, I think it can't correctly detect no-op rebase still)

**Pull request merged**